### PR TITLE
[SERVICES-2616] Update transaction generation for Router SC

### PR DESCRIPTION
**Pull request by mad2sm0key**
## Reasoning
- use SmartContractTransactionsFactory for generating transactions for Router SC
  
## Proposed Changes
- update `multiPairSwap` method in auto-router.transaction.service
- update methods for generating transactions in router.transaction.service to use SmartContractTransactionsFactory
- add sender parameter where needed  + update resolver methods
- fix unit tests

## How to test
- N/A

### CODE REVIEW

**Review by bogdan-rosianu — APPROVED** [review]

File: src/modules/router/specs/router.transactions.service.spec.ts [hunk 1]
```diff
@@ -76,23 +79,57 @@ describe('RouterService', () => {
...
+    it('should get upgrade pair transaction', async () => {
+        const service = module.get<RouterTransactionService>(
+            RouterTransactionService,
+        );
+
+        const transaction = await service.upgradePair(
+            senderAddress,
+            'WEGLD-123456',
+            'USDC-123456',
+            [10000, 10000],
```
Comment (claudiulataretu) [inline src/modules/router/specs/router.transactions.service.spec.ts:118]: use [300, 100] to have values close to real scenario

File: src/modules/router/services/router.transactions.service.ts [hunk 2]
```diff
@@ -77,25 +82,29 @@ export class RouterTransactionService {
             throw new Error('Pair does not exist');
         }
 
-        const contract = await this.mxProxy.getRouterSmartContract();
         const endpointArgs: TypedValue[] = [
             BytesValue.fromUTF8(firstTokenID),
             BytesValue.fromUTF8(secondTokenID),
+            new AddressValue(Address.newFromBech32(sender)),
```
Comment (claudiulataretu) [inline src/modules/router/services/router.transactions.service.ts:88]: this value is not actual accurate. Better would be to get it from the existing pair by the pair abi service and use that value here

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: src/modules/router/specs/router.transactions.service.spec.ts (hunk 1 above)
Comment (mad2sm0key) [reply src/modules/router/specs/router.transactions.service.spec.ts]: done

File: src/modules/router/services/router.transactions.service.ts (hunk 2 above)
Comment (mad2sm0key) [reply src/modules/router/services/router.transactions.service.ts]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**